proposals/0351-regex-builder.md

```diff
@@ -172,17 +169,25 @@ Regex builder addresses all of textual regexes' shortcomings presented in the [M
 
 ### `RegexComponent` protocol
 
-One of the goals of the regex builder DSL is allowing the developers to easily compose regexes from common currency types and literals, or even define custom patterns to use for matching. We introduce `RegexComponent`, a protocol that unifies all types that can represent a component of a regex.
+One of the goals of the regex builder DSL is allowing the developers to easily compose regexes from common currency types and literals, or even define custom patterns to use for matching. We introduce `RegexComponent` in the Swift Standard Library, a protocol that unifies all types that can represent a component of a regex. Since regexes are composable, the `Regex` type itself conforms to `RegexComponent`.
```

Comment:
I don't know if we've formally laid out what is stdlib vs the `Swift` module, and whether the regex builders modules is considered stdlib (and if not, should be it be considered as?).

```suggestion
One of the goals of the regex builder DSL is allowing the developers to easily compose regexes from common currency types and literals, or even define custom patterns to use for matching. We introduce `RegexComponent` in the `Swift` module, a protocol that unifies all types that can represent a component of a regex. Since regexes are composable, the `Regex` type itself conforms to `RegexComponent`.
```

Reply:
It might make sense to mention the new module first too, maybe a first section like:

### `RegexBuilder` module

...

Reply:
FWIW, I say "stdlib" or "stdlib proper" as a stand-in for the `Swift` module; I'm much more guilty than most. I honestly don't know what terminology to use. Since this proposal is one of the first to need that distinction, let's be as clear as we can be.

Reply:
I think that it will take a little time to iron out the nomenclature here, so let's go with "`Swift` module" for now (since we want to unblock this quickly); it is at least precise and unambiguous.

Reply:
FWIW, saying "the `Swift` module` means you would be able to do a qualified name lookup such as `Swift.Regex`, which doesn't work in our case. Nonetheless, I've used "the `Swift` module" for now.